src/slowmokit/methods/neighbors/distance_metric/distance_metric.cpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * @file methods/neighbors/distance_metric/distance_metric.hpp
+ *
+ * Easy include to calculate distance metrics
+ */
+
+#include "distance_metric.hpp"
+
+template<class T>
+DistanceMetric<T>::DistanceMetric(std::vector<T> &x, std::vector<T> &t)
+{
+  this->x = x;
+  this->y = y;
+}
+
+template<class T>
+double DistanceMetric<T>::euclidean(std::vector<T> &x, std::vector<T> &y)
```

Comment:
why are u passing x and y vectors each time when you've already stored them in the class constructor